app/routes/yara_rules.py

```diff
@@ -562,3 +567,101 @@ def delete_all_inactive_yara_rules():
+
+    current_entity = yara_rule.Yara_rule(
+        state=yara_revision_dict['state'],
+        name=yara_revision_dict['name'],
+        description=yara_revision_dict['description'],
+        references=yara_revision_dict['references'],
+        category=yara_revision_dict['category'],
+        condition=re.sub('^condition:\\n\\t', '', yara_revision_dict['condition']),
+        strings=re.sub('^strings:\\n\\t', '', yara_revision_dict['strings']),
+        eventid=temp_sig_id,
+        id=yara_rule_id,
+        creation_date=yara_revision_dict['creation_date'],
+        modified_user_id=current_user.id,
+        last_revision_date=datetime.datetime.now(),
+        owner_user_id=yara_revision_dict['owner_user']['id'] if yara_revision_dict['owner_user'] else None,
+        revision=current_entity.revision + 1,
+        imports=yara_revision_dict['imports'],
+        active=yara_revision_dict['active'],
+        mitre_techniques=yara_revision_dict['mitre_techniques'],
+        mitre_tactics=yara_revision_dict['mitre_tactics'],
+        files=yara_revision_dict['files'],
+    )
+
+    mitre_techniques = Cfg_settings.get_setting("MITRE_TECHNIQUES").split(",")
+    matches = [technique for technique in current_entity.mitre_techniques if technique not in mitre_techniques]
+    if matches:
+        raise (Exception(
+            "The following techniques were not found in the configuration: %s. Check 'MITRE_TECHNIQUES' on the settings page" % (
+                matches)))
+
+    mitre_tactics = Cfg_settings.get_setting("MITRE_TACTICS").split(",")
+    matches = [tactic for tactic in current_entity.mitre_tactics if tactic not in mitre_tactics]
+    if matches:
+        raise (Exception(
+            "The following tactics were not found in the configuration: %s. Check 'MITRE_TACTICS' on the settings page" % (
+                matches)))
+
+    db.session.merge(current_entity)
+    db.session.commit()
+
+    dirty = False
+    for name, value_dict in yara_revision_dict['metadata_values'].iteritems():
+        if not name or not value_dict:
+            continue
+
+        m = db.session.query(MetadataMapping).join(Metadata, Metadata.id == MetadataMapping.metadata_id).filter(
+            Metadata.key == name).filter(Metadata.artifact_type == ENTITY_MAPPING["SIGNATURE"]).filter(
+            MetadataMapping.artifact_id == current_entity.id).first()
+        if m:
+            m.value = value_dict["value"]
+            db.session.add(m)
+            dirty = True
+        else:
+            m = db.session.query(Metadata).filter(Metadata.key == name).filter(
+                Metadata.artifact_type == ENTITY_MAPPING["SIGNATURE"]).first()
+            db.session.add(MetadataMapping(value=value_dict["value"], metadata_id=m.id, artifact_id=current_entity.id,
+                                           created_user_id=current_user.id))
+            dirty = True
+
+    if dirty:
+        db.session.commit()
```

Comment:
Copy/Pasted this metadata logic from the `update` method... let me know if it's not needed here